rcl_action/src/rcl_action/goal_handle.c

```diff
@@ -0,0 +1,155 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include "rcl_action/goal_handle.h"
+
+#include "rcl/rcl.h"
+#include "rcl/error_handling.h"
+
+typedef struct rcl_action_goal_handle_impl_t
+{
+  rcl_action_goal_info_t info;
+  rcl_action_goal_state_t state;
+  rcl_allocator_t allocator;
+} rcl_action_goal_handle_impl_t;
+
+rcl_action_goal_handle_t
+rcl_action_get_zero_initialized_goal_handle(void)
+{
+  static rcl_action_goal_handle_t null_handle = {0};
+  return null_handle;
+}
+
+rcl_ret_t
+rcl_action_goal_handle_init(
+  rcl_action_goal_handle_t * goal_handle,
+  rcl_action_goal_info_t * goal_info,
+  const rcl_allocator_t allocator)
```

Comment:
@jacobperron meta (feel free to ignore): isn't marking `allocator` as `const` a bit misleading? I mean, we're not mutating the struct itself but we are mutating its state by allocating memory.

Reply:
Fixed in 29f0183